browser/src/neovim/Session.ts

```diff
@@ -80,11 +92,26 @@ export class Session extends EventEmitter {
         })
     }
 
+    // Neovim does not error if it is unable to get lines instead it returns an array
+    // of type [1, "an error message"] **on Some occasions**, we only check the first on the assumption that
+    // that is where the number is placed by neovim
+    public isNeovimError = <T>(val: T, methodName: string) => {
+        switch (methodName) {
+            case "nvim_buf_get_lines":
```

Comment:
It doesn't really make sense to need to check this here. The `Session` layer should be agnostic of the specific methods available on nvim (you could think of it is a _general_ msgpack Session, not just for Neovim).

I was hoping that, at either this layer or `NeovimInstance`, we could find a _general_ way to detect errors. If that's not the case, it'd be better to have the error handling local in the `getLines` method as we did before.